hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountInfoHandler.java

```diff
@@ -379,13 +378,13 @@ private void addNodeStakeMeta(
     }
 
     private boolean mayHavePendingReward(Account account) {
-        return account.stakedNumber() < 0 && !account.declineReward();
+        return account.hasStakedNodeId() && !account.declineReward();
     }
 
     private long getStakedNodeAddressBookId(Account account) {
-        if (account.stakedNumber() >= 0) {
+        if (!account.hasStakedNodeId()) {
             throw new IllegalStateException("Account is not staked to a node");
         }
-        return -account.stakedNumber() - 1;
+        return -account.stakedNodeId() - 1;
```

Comment:
shouldn't this just be `account.stakedNodeId()` ?

Reply:
Updated. Thanks.